server/sonar-server/src/main/java/org/sonar/server/computation/component/ComponentImpl.java

```diff
@@ -126,7 +127,7 @@ private Builder(ScannerReport.Component component) {
       checkNotNull(component);
       this.type = convertType(component.getType());
       this.name = checkNotNull(component.getName());
```

Comment:
component.getName() can not be null, so `checkNotNull` is useless. It could be replaced by the check `checkNotEmpty`.